### PR TITLE
Added fresh method to entity class

### PR DESCRIPTION
**Pull request by Gman98ish**
This method clones an entity, and all the entities it holds.

This is really useful if you need a copy of the entity you can safely mutate without affecting the original

### CODE REVIEW

File: src/Entity.php [hunk 1]
```diff
@@ -160,6 +160,34 @@ public function toArray($map = [])
         return $arr;
     }
 
+    /**
+     * Clones an entity and all nested entities
+     * Useful if you need to operate on a copy of an entity
+     * and not worry about mutating anything in the original
+     * @return Entity
+     */
+    public function fresh()
```
Comment (Gman98ish) [inline src/Entity.php]: @phily245 made a good point about the naming of this, `copy` would make more sense here

Comment (Gman98ish) [reply src/Entity.php]: Also need documenting in the README or something

Comment (tylerwoonton) [reply src/Entity.php]: Oh, that's true. I'm thinking the Eloquent usage of fresh is for a new instance of the model. Whereas this is a copy of the current entity?

Comment (Gman98ish) [reply src/Entity.php]: yes

**Pull request merged**